### ISSUE TITLE
Does spark-iceberg support create or drop tag/branch?

### ISSUE THREAD

**Issue by lexluo09** [apache/iceberg#6902]
### Query engine

Spark

### Question

Does spark-iceberg support create or drop tag/branch
1、create tag
ALTER TABLE demo.nyc.taxis
CREATE TAG historical_annual_snapshot
AS OF VERSION 100
RETAIN 12 MONTHS

2、create branch
ALTER TABLE demo.nyc.taxis
CREATE BRANCH test_branch AS OF VERSION 100 WITH SNAPSHOT RETENTION 5 SNAPSHOTS 7 DAYS
RETAIN 12 MONTHS

3、drop tag/branch
ALTER TABLE table DROP {BRANCH | TAG} reference

Does spark-iceberg support create or drop tag/branch
1、create tag
ALTER TABLE demo.nyc.taxis
CREATE TAG historical_annual_snapshot
AS OF VERSION 100
RETAIN 12 MONTHS

2、create branch
ALTER TABLE demo.nyc.taxis
CREATE BRANCH test_branch AS OF VERSION 100 WITH SNAPSHOT RETENTION 5 SNAPSHOTS 7 DAYS
RETAIN 12 MONTHS

3、drop tag/branch
ALTER TABLE table DROP {BRANCH | TAG} reference

My operation failed !

![image](https://user-images.githubusercontent.com/39718951/220599587-c4eadc32-7c1f-4a84-87f9-922eac1a6f86.png)

4、
CALL demo.system.create_tag(table => 'demo.nyc.taxis', snapshot_id => 7406989859988178505, tag_name => 'testTag');

![image](https://user-images.githubusercontent.com/39718951/220599826-5f5b8089-e413-4141-a9ac-e2efbd44a51d.png)

**Comment by lexluo09** [comment]
Can someone take a look? Thanks

**Comment by nastra** [comment]
@lexluo09 yes this is [supported](https://github.com/apache/iceberg/projects/4) but it has not been released yet. It will be released in the next few weeks as part of Iceberg 1.2.0.

**Comment by lexluo09** [comment]
> the
Thank you very much for your answer

**Comment by DavidCGarcia89** [comment]
I have also the same problem in AWS. I'm trying to execute a job with Scala trying to create a TAG but get this error:

![image](https://github.com/apache/iceberg/assets/20771695/74cfe18f-9eeb-4784-8c3b-73265ccf886f)

**Comment by bhaskar-pv** [comment]
The issue exists yet

**Comment by ldocao** [comment]
Hi,

I'm also encountering the same issue while running on iceberg 1.5.2 locally with pyspark 3.5.2

```
E               pyspark.errors.exceptions.captured.ParseException:
E               [PARSE_SYNTAX_ERROR] Syntax error at or near 'CREATE'.(line 1, pos 54)
E
E               == SQL ==
E               ALTER TABLE iceberg_test.nyc_taxi.fhvhv_trips_iceberg CREATE TAG `2024-01-01` AS OF VERSION 8461765733328903298

```

Here is my setup (iceberg 1.5.2 with spark 3.5.2):

```python
WAREHOUSE_PATH = "./warehouse"
catalog = "test_iceberg"
spark_builder = (spark_builder
        .config(f"spark.sql.catalog.{catalog}", "org.apache.iceberg.spark.SparkCatalog")
        .config(f"spark.sql.catalog.{catalog}.type", "hadoop")
        .config(f"spark.sql.catalog.{catalog}.warehouse", WAREHOUSE_PATH)
        .config(f"spark.jars.packages", "org.apache.iceberg:iceberg-spark-runtime-3.5_2.12:1.5.2")
        .config(f"spark.sql.catalog.{catalog}.default.write.metadata-flush-after-create", "true") 
    ) 
spark = spark_builder.getOrCreate()
spark.sql(f"""
        CREATE TABLE {full_name} (
            hvfhs_license_num STRING,
            dispatching_base_num STRING,
            originating_base_num STRING,
            ...
        )
        USING iceberg;""") 
#i write into iceberg here without error
spark.sql("ALTER TABLE iceberg_test.nyc_taxi.fhvhv_trips_iceberg CREATE TAG `2024-01-01` AS OF VERSION 8461765733328903298") #raises [PARSE_SYNTAX_ERROR] Syntax error at or near 'CREATE'.(line 1, pos 54)
```

**Comment by nastra** [comment]
@ldocao it looks like you're missing the `spark.sql.extensions=org.apache.iceberg.spark.extensions.IcebergSparkSessionExtensions` config. Without the config the custom SQL won't work
